scipy/linalg/tests/test_special_matrices.py

```diff
@@ -640,3 +642,45 @@ def test_fiedler_companion():
     fc = fiedler_companion([1., -16., 86., -176., 105.])
     assert_array_almost_equal(eigvals(fc),
                               np.array([7., 5., 3., 1.]))
+
+
+def test_convolution_matrix():
+    'test convolution_matrix vs. numpy.convolve for various parameters'
+
+    def test_vector(n, cpx):
+        'make a complex or real test vector of length n'
+        if cpx:
+            return np.random.normal(size=(n, 2))\
+                .astype(np.float64).view(np.complex128).ravel()
+        else:
+            return np.random.normal(size=(n,))
+
+    # first arg must be a 1d array, otherwise ValueError
+    with assert_raises(ValueError):
+        convolution_matrix(1, 4)
+
+    # mode must be in ('full','valid','same')
+    with assert_raises(ValueError):
```

Comment:
This is testing something else

Reply:
That's unnecessary granularity. And also if there is no scope nesting it's just a preference. If we write one test per assertion we will have millions of tests.

Reply:
See examples in the rest of the file for more examples.

Reply:
Unfortunately, there is a lot of code in our codebase that does not conform to what we now consider best practices.  It is our job as reviewers to ensure the *new* code does so.